packages/cli-kit/src/private/node/api/graphql.ts

```diff
@@ -65,3 +79,11 @@ function errorHandler<T>(api: string): (error: unknown) => Error | unknown {
     }
   }
 }
+
+export function handleDeprecations<T>(response: GraphQLResponse<T>) {
+  const deprecations = (response.extensions as WithDeprecations)?.deprecations
```

Comment:
`response.extensions` breaks the service-agnostic nature of the APIs in this module. What about the following:
- If the schema is specific to the Partners' API, we can have a version of this function, `partnersGraphQLRequest` that lives in the `@shopify/app` package (which is the only one interacting with partners).
- If the schema is specific to one of the queries and mutations, I'd move the code to where that request is happening.

Reply:
We intended for the schema to be generalized beyond Partners API so other APIs could also return deprecations if necessary with the same mechanism. Maybe I can pass an optional `handleDeprecations` function to `graphqlRequest` for the caller to decide?